From 13babb1a3ea09a39280861f93df1845b26d77ca1 Mon Sep 17 00:00:00 2001
From: Jerome Forissier <jerome@forissier.org>
Date: Wed, 22 Apr 2020 18:46:30 +0200
Subject: [PATCH 4/6] core: move for_each_early_ta() macro to
 <kernel/early_ta.h>

Move the for_each_early_ta() macro out of early_ta.c so that it can be
used in other parts of the code (pseudo TAs for instance).

Signed-off-by: Jerome Forissier <jerome@forissier.org>
Signed-off-by: Maxim Uvarov <maxim.uvarov@linaro.org>
---
 core/arch/arm/include/kernel/early_ta.h | 8 ++++++++
 core/arch/arm/kernel/early_ta.c         | 7 -------
 2 files changed, 8 insertions(+), 7 deletions(-)

diff --git a/core/arch/arm/include/kernel/early_ta.h b/core/arch/arm/include/kernel/early_ta.h
index 9b0c6288..0ad8312c 100644
--- a/core/arch/arm/include/kernel/early_ta.h
+++ b/core/arch/arm/include/kernel/early_ta.h
@@ -6,8 +6,10 @@
 #define KERNEL_EARLY_TA_H
 
 #include <compiler.h>
+#include <kernel/linker.h>
 #include <stdint.h>
 #include <tee_api_types.h>
+#include <util.h>
 
 struct early_ta {
 	uint32_t flags;
@@ -19,5 +21,11 @@ struct early_ta {
 
 #define __early_ta __section(".rodata.early_ta" __SECTION_FLAGS_RODATA)
 
+#define for_each_early_ta(_ta) \
+	for (_ta = &__rodata_early_ta_start; _ta < &__rodata_early_ta_end; \
+	     _ta = (const struct early_ta *)				   \
+		   ROUNDUP((vaddr_t)_ta + sizeof(*_ta) + _ta->size,	   \
+			   __alignof__(struct early_ta)))
+
 #endif /* KERNEL_EARLY_TA_H */
 
diff --git a/core/arch/arm/kernel/early_ta.c b/core/arch/arm/kernel/early_ta.c
index 05965cbc..bfec58cd 100644
--- a/core/arch/arm/kernel/early_ta.c
+++ b/core/arch/arm/kernel/early_ta.c
@@ -5,7 +5,6 @@
 #include <crypto/crypto.h>
 #include <initcall.h>
 #include <kernel/early_ta.h>
-#include <kernel/linker.h>
 #include <kernel/user_ta.h>
 #include <kernel/user_ta_store.h>
 #include <stdio.h>
@@ -21,12 +20,6 @@ struct user_ta_store_handle {
 	z_stream strm;
 };
 
-#define for_each_early_ta(_ta) \
-	for (_ta = &__rodata_early_ta_start; _ta < &__rodata_early_ta_end; \
-	     _ta = (const struct early_ta *)				   \
-		   ROUNDUP((vaddr_t)_ta + sizeof(*_ta) + _ta->size,	   \
-			   __alignof__(struct early_ta)))
-
 static const struct early_ta *find_early_ta(const TEE_UUID *uuid)
 {
 	const struct early_ta *ta;
-- 
2.17.1

